skyportal/handlers/api/photometry.py

```diff
@@ -649,6 +668,71 @@ def get_stream_ids(self):
         stream_ids = list(set(stream_ids))
         return stream_ids
 
+    def add_external_photometry(self, json, user):
+        """
+        ---
+        description: Upload photometry
+        tags:
+          - photometry
+        parameters:
+          - in: path
+            name: json
+            required: True
+            schema:
+              oneOf:
+                - $ref: "#/components/schemas/PhotMagFlexible"
+                - $ref: "#/components/schemas/PhotFluxFlexible"
+          - in: path
+            name: user
+            required: True
+            schema: SingleUser
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            ids:
+                              type: array
+                              items:
+                                type: integer
+                              description: List of new photometry IDs
+                            upload_id:
+                              type: string
+                              description: |
+                                Upload ID associated with all photometry points
+                                added in request. Can be used to later delete all
+                                points in a single request.
+        """
+
+        group_ids = self.get_group_ids(json=json, user=user)
+        stream_ids = self.get_stream_ids(json=json)
+        df, instrument_cache = self.standardize_photometry_data(json=json)
+
+        # This lock ensures that the Photometry table data are not modified in any way
+        # between when the query for duplicate photometry is first executed and
+        # when the insert statement with the new photometry is performed.
+        # From the psql docs: This mode protects a table against concurrent
+        # data changes, and is self-exclusive so that only one session can
+        # hold it at a time.
+        DBSession().execute(
+            f'LOCK TABLE {Photometry.__tablename__} IN SHARE ROW EXCLUSIVE MODE'
+        )
+        try:
+            ids, upload_id = self.insert_new_photometry_data(
+                df, instrument_cache, group_ids, stream_ids, user=user
+            )
+        except ValidationError as e:
+            return self.error(e.args[0])
```

Comment:
is the table gauranteed to be unlocked? What if the exception happens? I dont see an unlock explictly done here.

Reply:
This is just the same as what is already in POST.

Reply:
I'd like @stefanv to weigh in here since a DB lock seems very risky to me (it might be fine but I don't know now this works so I want me make sure).  It's possible the POST needs to get reworked if this is a concern.